networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AbstractGossipManager.java

```diff
@@ -57,48 +56,25 @@ protected AbstractGossipManager(
             topicName,
             gossipType,
             maxMessageSize);
-    this.channel = gossipNetwork.subscribe(topicHandler.getTopic(), topicHandler);
     this.gossipEncoding = gossipEncoding;
-    this.publisher = publisher;
-
-    this.subscriberId = publisher.subscribe(this::publishMessage);
   }
 
-  protected AbstractGossipManager(
-      final RecentChainData recentChainData,
-      final GossipTopicName topicName,
-      final AsyncRunner asyncRunner,
-      final GossipNetwork gossipNetwork,
-      final GossipEncoding gossipEncoding,
-      final ForkInfo forkInfo,
-      final OperationProcessor<T> processor,
-      final GossipPublisher<T> publisher,
-      final SszSchema<T> gossipType,
-      final int maxMessageSize) {
-    this(
-        recentChainData,
-        topicName.toString(),
-        asyncRunner,
-        gossipNetwork,
-        gossipEncoding,
-        forkInfo,
-        processor,
-        publisher,
-        gossipType,
-        maxMessageSize);
+  protected void publishMessage(T message) {
+    channel.ifPresent(c -> c.gossip(gossipEncoding.encode(message)));
   }
 
-  protected void publishMessage(T message) {
-    final Bytes data = gossipEncoding.encode(message);
-    channel.gossip(data);
+  @Override
+  public void subscribe() {
+    if (channel.isEmpty()) {
+      channel = Optional.of(gossipNetwork.subscribe(topicHandler.getTopic(), topicHandler));
+    }
   }
 
   @Override
-  public void shutdown() {
-    if (shutdown.compareAndSet(false, true)) {
-      // Close gossip channels
-      channel.close();
-      publisher.unsubscribe(subscriberId);
+  public void unsubscribe() {
+    if (channel.isPresent()) {
+      channel.get().close();
+      channel = Optional.empty();
```

Comment:
could lead to concurrency fun with the subscribe, do they need to be synchronized?

Reply:
They're synchronized back up at the `GossipForkManager` level.

Reply:
hmm, except potentially for the ones that use a gossip publisher... Will go check those.